cli/src/util/api.rs

```diff
@@ -0,0 +1,77 @@
+use global_error::prelude::*;
+use regex::Regex;
+use url::Url;
+
+/// Normalize an API endpoint to a standardized URL.
+///
+/// This allows us to pass shorthand URLs like:
+///
+/// * api.staging2.gameinc.io -> https://api.staging2.gameinc.io
+/// * 127.0.0.1:8080 -> http://127.0.0.1:8080
+pub fn normalize_api_endpoint(endpoint: &str) -> GlobalResult<String> {
+	// Attempt to parse URL
+	let url = match Url::parse(endpoint) {
+		Ok(url) => url,
+		Err(url::ParseError::RelativeUrlWithoutBase) => {
+			// No scheme was provided, determine if to use HTTP or HTTPS based on if the URL is
+			// localhost.
+			let localhost_regex = Regex::new(r"^(127\.0\.0\.1|\[::1\])(:\d+)?$")?;
```

Comment:
Should this also catch `localhost` and LAN machines `192.x.x.x` or WAN addresses?

Going a bit deeper, should this maybe try hitting `https` and falling back to `http` in cases where it can't get through? (probably not, this takes time and it's probably fine to be general)

Will us guessing on a user's behalf have the chance to cause mishaps that aren't easy for them to diagnose? Should we maybe add a println to say `translating URL to http://127.0.0.1` or something like that?